### PR TITLE
Move proxy code to an example

### PR DESCRIPTION
**Pull request by YorikSar**
It is an example, after all, and it feels like it belongs in a separate space.

cc @jneem

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -339,121 +316,8 @@ impl<W: Write> NixWrite<W> {
     }
 }
 
-impl<R: Read, W: Write> NixProxy<R, W> {
-    // Wait for an initialization message from the client, and perform
-    // the version negotiation.
-    //
-    // Returns the client version.
-    pub fn handshake(&mut self) -> Result<u64> {
```
Comment (jneem) [inline src/lib.rs:347]: I agree with most of this PR, and clearly the public API here needs some thought, but we're actually using this function [elsewhere](https://github.com/tweag/remote-execution-nix/blob/d7b21a7aa54b6ed5482ad2b09662d880fef2fc6d/crates/proxy/src/main.rs#L530) so I think it makes sense to have it in the library.

**Review by jneem — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (YorikSar) [reply src/lib.rs:347]: Oh, I see you’re using NixProxy and some of its methods there. I’ll think about extracting some reusable API subset from it.

**Comment by YorikSar** [comment]
I've started extracting all interfaces in https://github.com/tweag/nix-remote-rust/pull/8 keeping in mind [remote-execution-nix](https://github.com/tweag/remote-execution-nix/blob/d7b21a7aa54b6ed5482ad2b09662d880fef2fc6d/crates/proxy/src/main.rs#L530) requirements.

**Comment by YorikSar** [comment]
Closed in favour of #7

**Pull request closed without merging**